hal/Cargo.toml

```diff
@@ -1,227 +1,342 @@
+fugit = {version = "0.3", optional = true}
+jlink_rtt = {version = "0.2", optional = true}
+rtic-monotonic = {version = "1.0", optional = true}
+usb-device = {version = "0.2", optional = true}
+
+#===============================================================================
+# PACs
+#===============================================================================
 
-[dependencies.jlink_rtt]
-optional = true
-version = "0.2"
-
-[dependencies.num-traits]
-default-features = false
-version = "0.2.14"
-
-[dependencies.usb-device]
-version = "0.2"
-optional = true
-
-[dependencies.rtic-monotonic]
-optional = true
-version = "1.0"
-
-[dependencies.fugit]
-optional = true
-version = "0.3"
-
-[dependencies.void]
-default-features = false
-version = "1.0"
-
-# Each of the supported chips is listed as an optional dependency here.
-# This makes it available when the corresponding feature name is referenced.
-# We use a feature named "samdFOO" to pull in the dependency named "atsamdFOO"
-[dependencies.atsamd11c]
-path = "../pac/atsamd11c"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd11d]
-path = "../pac/atsamd11d"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd21e]
-path = "../pac/atsamd21e"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd21g]
-path = "../pac/atsamd21g"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd21j]
-path = "../pac/atsamd21j"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd51g]
-path = "../pac/atsamd51g"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd51j]
-path = "../pac/atsamd51j"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd51n]
-path = "../pac/atsamd51n"
-version = "0.12"
-optional = true
-
-[dependencies.atsamd51p]
-path = "../pac/atsamd51p"
-version = "0.12"
-optional = true
-
-[dependencies.atsame51g]
-path = "../pac/atsame51g"
-version = "0.12"
-optional = true
-
-[dependencies.atsame51j]
-path = "../pac/atsame51j"
-version = "0.12"
-optional = true
-
-[dependencies.atsame51n]
-path = "../pac/atsame51n"
-version = "0.12"
-optional = true
-
-[dependencies.atsame53j]
-path = "../pac/atsame53j"
-version = "0.12"
-optional = true
-
-[dependencies.atsame53n]
-path = "../pac/atsame53n"
-version = "0.12"
-optional = true
-
-[dependencies.atsame54n]
-path = "../pac/atsame54n"
-version = "0.12"
-optional = true
-
-[dependencies.atsame54p]
-path = "../pac/atsame54p"
-version = "0.12"
-optional = true
-
-[dependencies.embedded-sdmmc]
-version = "0.3"
-optional = true
+# The peripheral access crates (PAC) for each supported chip variant are listed
+# here. Each is an optional dependency, which means each PAC name also appears
+# as a Cargo feature. Users should not select PAC features manually. Instead,
+# users should specify a corresponding variant (see below). The variant features
+# will select the correct PAC, as well as other configuration features.
 
+atsamd11c = {version = "0.12", path = "../pac/atsamd11c", optional = true}
```

Comment:
I'm not sure whether the bump-crates workflow will like us changing the format of these, I guess we'll find out.

Reply:
I was wondering why the PAC dependencies were formatted that way. I found the spread out version to be very difficult to parse. I personally find it much easier to read when they are condensed to single lines.

However, I didn't consider that the specific syntax of the TOML file might be depended on elsewhere. That seems extremely brittle, though. Maybe we can find a better way? I found a [toml-cli](https://github.com/gnprice/toml-cli) project. Could we use that instead?

Reply:
You're right, this will break the workflow. Let me look at updating it. Maybe there's a less brittle way to do this.

Reply:
It looks like the `tomlkit` package can parse, modify and dump TOML files while preserving comments and formatting. Let me see if I can modify the scripts to use it.

Reply:
@sajattack, I updated the two scripts I found that depend on the formatting of the TOML files. Are those the only two scripts you know of? They are the only two I could find.

Could you have a look at the modifications and make sure I didn't accidentally change the behavior?